src/ice_shelf/MOM_ice_shelf_dynamics.F90

```diff
@@ -897,12 +908,71 @@ subroutine update_ice_shelf(CS, ISS, G, US, time_step, Time, ocean_mass, coupled
 
   if (update_ice_vel) then
     call ice_shelf_solve_outer(CS, ISS, G, US, CS%u_shelf, CS%v_shelf,CS%taudx_shelf,CS%taudy_shelf, iters, Time)
+    CS%elapsed_velocity_time = 0.0
   endif
 
 ! call ice_shelf_temp(CS, ISS, G, US, time_step, ISS%water_flux, Time)
 
-  if (CS%elapsed_velocity_time >= CS%velocity_update_time_step) then
-    call enable_averages(CS%elapsed_velocity_time, Time, CS%diag)
+end subroutine update_ice_shelf
+
+subroutine volume_above_floatation(CS, G, ISS, vab)
+  type(ice_shelf_dyn_CS), intent(in) :: CS !< The ice shelf dynamics control structure
+  type(ocean_grid_type),  intent(in) :: G  !< The grid structure used by the ice shelf.
+  type(ice_shelf_state),  intent(in) :: ISS !< A structure with elements that describe
+                                            !! the ice-shelf state
+  real, intent(out) :: vab !< area integrated volume above floatation
+  real, dimension(SZI_(G),SZJ_(G))  :: vab_cell !< cell-wise volume above floatation [Z ~> m}
```

Comment:
As the code is written, it appears that the units of `vab_cell` are [m3], and not [Z ~> m].  Also, please add the units to the description of `vab`, which I suspect are also [m3].